publisher/publisher.go

```diff
@@ -92,9 +92,10 @@ func (pub *Impl) SubmitToCT(ctx context.Context, der []byte) error {
 
 	chain := append([]ct.ASN1Cert{der}, pub.issuerBundle...)
 	for _, ctLog := range pub.ctLogs {
-		ctx, cancel := context.WithTimeout(context.Background(), pub.submissionTimeout)
+		localCtx, cancel := context.WithTimeout(ctx, pub.submissionTimeout)
+		// since there are possible error cases defer instead of explicitly cancelling
```

Comment:
if https://github.com/letsencrypt/boulder/pull/1708/files#diff-dc679ea47245363b29b155017fc06ac8R127 isn't using localCtx no reason not to cancel it now, right?